### PR TITLE
Add build definition for notcurses 2.2.4

### PR DESCRIPTION
**Pull request by SystemNeoSpl**
Inital wrap files for notcurses-2.2.4.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,320 @@
+project('notcurses', ['c', 'cpp'],
+        default_options:
+          ['c_std=gnu11',
+           'cpp_std=gnu++17',
+           'default_library=both',
+           'c_args=-Wall -Wextra -W -Wshadow -Wformat -fexceptions',
+           'cpp_args=-Wall -Wextra -W -Wshadow -Wformat -fexceptions'
```
Comment (jpakkane) [inline meson.build]: These should not be here. `-Wall`and `-Wextra` should be enabled with `warning_level=3` instead. The other flags should be added with `add_project_args` instead after testing that they are supported with `get_supported_arguments`.

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+  error('Could not find uniwbrk.h from GNU libunistring. STOP.')
+else
+  deps += cc.find_library('unistring', required : true)
+endif
+
+if get_option('use_qrcodegen')
+  deps += cc.find_library('libqrcodegen', required : true)
+endif
+
+deps += cc.find_library('rt')
```
Comment (jpakkane) [inline meson.build]: This should have `required: false` for portability. Many platforms do not have `rt`, it is a part of libc instead.

File: upstream.wrap [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+source_url = https://github.com/dankamongmen/notcurses/archive/refs/tags/v2.2.4.tar.gz
+source_filename = v2.2.4.tar.gz
+source_hash = f062b4a86ab61e800c8754cd36af557f41ac8a015e3fe3921dab1ed37c665394
+
+[provide]
+notcurses_core = notcurses_core_dep
+notcurses = notcurses_dep
+ncpp = ncpp_dep
+
+# ft: dosini
```
Comment (jpakkane) [inline upstream.wrap]: This line should not be here.

**Review by jpakkane — CHANGES_REQUESTED** [review]

File: meson.build [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+        version: '2.2.4',
+        license: 'Apache-2.0',
+        meson_version: '>=0.57.1')
+
+fs = import('fs')
+
+cc = meson.get_compiler('c')
+cxx = meson.get_compiler('cpp')
+
+compflags = ['-Wshadow','-Wformat','-fexceptions', '-Og', '-D_FORTIFY_SOURCE=2']
```
Comment (jpakkane) [inline meson.build:15]: Please remove `-Og` and `-D_FORTIFY_SOURCE=2`. As mentioned earlier they should not be set from within `meson.build` files.

File: upstream.wrap [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+directory = notcurses-2.2.4
+
+source_url = https://github.com/dankamongmen/notcurses/archive/refs/tags/v2.2.4.tar.gz
+source_filename = v2.2.4.tar.gz
+source_hash = f062b4a86ab61e800c8754cd36af557f41ac8a015e3fe3921dab1ed37c665394
+
+[provide]
+notcurses_core = notcurses_core_dep
+notcurses = notcurses_dep
+ncpp = ncpp_dep
```
Comment (jpakkane) [inline upstream.wrap:12]: I don't have this handy for testing, but just to be sure, these are the pkg-config names this project provides, right?

Comment (dankamongmen) [reply upstream.wrap:12]: they are not (package author here). the package names are:

* `notcurses-core`
* `notcurses`
* `notcurses++`

Please let me know if i can be of any help.

**Comment by jpakkane** [comment]
Closing this as individual wrap repos are no longer used. Please refile this against [the main repo instead](https://github.com/mesonbuild/notcurses/pull/2#discussion_r659231884). Thanks.

**Pull request closed without merging**